logserver.go

```diff
@@ -79,9 +81,18 @@ func serveFile(w http.ResponseWriter, r *http.Request, path string) {
 	// for anti-XSS belt-and-braces, set a very restrictive CSP
 	w.Header().Set("Content-Security-Policy", "default-src: none")
 
-	// if it's a directory, serve a listing
+	// if it's a directory, serve a listing or a tarball
 	if d.IsDir() {
-		log.Println("Serving", path)
+		format, _ := r.URL.Query()["format"]
+		if len(format) == 1 && format[0] == "tar.gz" {
+			log.Println("Serving tarball of", path)
+			err := serveTarball(w, r, path)
+			if err != nil {
+				log.Println("Error", err)
```

Comment:
think you need to call [`http.Error()`](https://pkg.go.dev/net/http#Error) in this case to make sure we serve a response of some sort if there is an error before we serve anything.

(also, `serveTarball` should probably not return an error if an error occurs once streaming has started.)